django/thunderstore/repository/models/cache.py

```diff
@@ -11,6 +11,42 @@
 from thunderstore.core.mixins import S3FileMixin
 
 
+class APIExperimentalPackageIndexCache(S3FileMixin):
+    @classmethod
+    def get_latest(cls) -> Optional["APIExperimentalPackageIndexCache"]:
+        return cls.objects.active().order_by("-last_modified").first()
+
+    @classmethod
+    def update(cls, content: bytes) -> "APIExperimentalPackageIndexCache":
+        gzipped = io.BytesIO()
+        with gzip.GzipFile(fileobj=gzipped, mode="wb") as f:
+            f.write(content)
+        timestamp = timezone.now()
+        file = ContentFile(
+            # TODO: This is immediately passed to BytesIO again, meaning
+            #       we're just wasting memory. Find a way to pass this to
+            #       the Django model without the inefficiency.
+            gzipped.getvalue(),
+            name=f"full-index-{timestamp.isoformat()}.json.gz",
+        )
+        return cls.objects.create(
+            data=file,
+            content_type="application/json",
+            content_encoding="gzip",
+            last_modified=timestamp,
+        )
+
+    @classmethod
+    def drop_stale_cache(cls):
+        latest = cls.get_latest()
+        if latest is None:
+            return
+        cutoff = latest.last_modified - timedelta(hours=1)
```

Comment:
What is the purpose for the cutoff? Why not just delete all but the latest? Or if the hour is significant, is it implicitly bound to the frequency on which the background task runs currently (but not necessarily in the future)?

Reply:
The cutoff is quite arbitrary, but the idea is that some clients might still be trying to access the older version for some time. Cleanup of the model here deletes the actual file from the storage & thus also CDN after all.

That being said, it should probably be increased a bit given how different the update schedule is for this one. Will make some change to it & include a comment on why it exists.

Reply:
Increased the duration to 3 hours and added a comment explaining why it exists